tools/precommit.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Samsung Electronics Co., Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import sys
+import os
+from common_py import path
+from common_py.system.filesystem import FileSystem as fs
+from common_py.system.executor import Executor as ex
+from common_py.system.platform import Platform
+
+TESTS=['host', 'rpi2', 'nuttx', 'darwin', 'misc']
```

Comment:
I made this `TESTS` variable just to separate the tests into multiple categories so that our 5 buildbots in travis can run it in parallell. I think it can be changed if the number of buildbots changes, or if we decide to separate the tests in different manner, etc. Moreover, it doesnt' fit in any of our build configuration variable: target-board, target-os, target-arch. 
I think it would be better to keep it here.